### PR TITLE
Allow users to send coinbases to a public key other than the proposer

### PR DESCRIPTION
**Pull request by imeckler**
This PR makes it possible to send coinbases to a public key other than that of the proposer. It adds a command line flag `-coinbase-receiver` to configure this. If not provided, it defaults to sending coinbases to the proposer.

### CODE REVIEW

File: src/lib/coda_state/snark_transition.ml [hunk 1]
```diff
@@ -43,7 +43,7 @@ module Poly = struct
     ; sok_digest: 'sok_digest
     ; supply_increase: 'amount
     ; ledger_proof: Proof.Stable.V1.t option
-    ; proposer: 'proposer_pk
+    ; coinbase_receiver: 'proposer_pk
```
Comment (wu-s-john) [inline src/lib/coda_state/snark_transition.ml]: NIT: can you change the name of the polymorphic type 'proposer_pk

File: src/lib/coda_state/snark_transition.mli [hunk 2]
```diff
@@ -14,7 +14,7 @@ module Poly : sig
     ; sok_digest: 'sok_digest
     ; supply_increase: 'amount
     ; ledger_proof: Proof.Stable.V1.t option
-    ; proposer: 'proposer_pk
+    ; coinbase_receiver: 'proposer_pk
```
Comment (wu-s-john) [inline src/lib/coda_state/snark_transition.mli]: change this as well

**Comment by deepthiskumar** [comment]
We are sending transaction fee as well to this new public key. Is that intended?

**Comment by imeckler** [comment]
@deepthiskumar It was intentional. The thinking is that whoever is running a node would want to transfer all their rewards to another key.

**Comment by imeckler** [comment]
@deepthiskumar It was intentional. The thinking is that whoever is running a node would want to transfer all their rewards to another key.

**Pull request merged**